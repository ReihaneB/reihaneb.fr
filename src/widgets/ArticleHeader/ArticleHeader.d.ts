import { ImageProps } from '@/components/Image/Image';

export interface ArticleHeaderProps {
  title: string;
  image: ImageProps;
}

export interface ArticleHeaderStoreProps {
  id: string;
}
