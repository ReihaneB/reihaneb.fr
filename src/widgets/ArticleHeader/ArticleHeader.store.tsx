import { getBlogArticle } from '@/api/getBlogArticle';

import ArticleHeader from './ArticleHeader';

import type { ArticleHeaderStoreProps } from './ArticleHeader.d';

async function ArticleHeaderStore({ id }: ArticleHeaderStoreProps) {
  const { title, image } = await getBlogArticle({ id });

  return (
    <ArticleHeader
      title={title}
      image={image}
    />
  );
}

export default ArticleHeaderStore;
