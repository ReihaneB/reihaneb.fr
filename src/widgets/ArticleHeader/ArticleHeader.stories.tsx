import type { Meta, StoryObj } from '@storybook/react';

import _16x9 from '@/styles/assets/placeholder/_16x9.png';

import ArticleHeader from './ArticleHeader';

const meta: Meta<typeof ArticleHeader> = {
  title: 'Widgets/ArticleHeader',
  component: ArticleHeader,
};

export default meta;
type Story = StoryObj<typeof ArticleHeader>;

const defaultArgs = {
  title: 'Article title',
  image: {
    src: _16x9,
    alt: 'Article image',
    width: 800,
    height: 400,
  },
};

export const Playground: Story = {
  args: {
    ...defaultArgs,
  },
};

export const NoImage: Story = {
  args: {
    title: 'Article title',
    image: undefined,
  },
};

export const LongTitle: Story = {
  args: {
    ...defaultArgs,
    title: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla nec purus feugiat, vestibulum mi id, lacinia orci. Sed nec metus nec justo lacinia ultricies. Nullam non',
  },
};

export const LongTitleNoImage: Story = {
  args: {
    title: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla nec purus feugiat, vestibulum mi id, lacinia orci. Sed nec metus nec justo lacinia ultricies. Nullam non',
    image: undefined,
  },
};
