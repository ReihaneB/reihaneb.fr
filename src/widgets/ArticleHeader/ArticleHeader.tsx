import { Fragment } from 'react';

import Image from '@/components/Image/Image';

import type { ArticleHeaderProps } from './ArticleHeader.d';
import styles from './ArticleHeader.module.css';

function ArticleHeader({
  title,
  image,
  ...rest
}: ArticleHeaderProps) {
  return (
    <Fragment {...rest}>
      <h1 className={styles.title}>
        {title}
      </h1>
      {image && (
        <Image
          src={image.src}
          alt={image.alt}
          width={image.width}
          height={image.height}
          priority
        />
      )}
    </Fragment>
  );
}

export default ArticleHeader;
