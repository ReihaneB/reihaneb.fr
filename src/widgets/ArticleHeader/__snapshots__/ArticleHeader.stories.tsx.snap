// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widgets/ArticleHeader LongTitle smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <h1 class="src-widgets-ArticleHeader-ArticleHeader-module__title--SK4qr">
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla nec purus feugiat, vestibulum mi id, lacinia orci. Sed nec metus nec justo lacinia ultricies. Nullam non
  </h1>
  <img alt="Article image"
       fetchpriority="high"
       width="800"
       height="400"
       decoding="async"
       data-nimg="1"
       class="src-components-Image-Image-module__image--gXjYq"
       srcset="/static/media/_16x9.d50e36d3.png?w=828&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 2x"
       src="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75"
       style="color: transparent;"
  >
</main>
`;

exports[`Widgets/ArticleHeader LongTitleNoImage smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <h1 class="src-widgets-ArticleHeader-ArticleHeader-module__title--SK4qr">
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla nec purus feugiat, vestibulum mi id, lacinia orci. Sed nec metus nec justo lacinia ultricies. Nullam non
  </h1>
</main>
`;

exports[`Widgets/ArticleHeader NoImage smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <h1 class="src-widgets-ArticleHeader-ArticleHeader-module__title--SK4qr">
    Article title
  </h1>
</main>
`;

exports[`Widgets/ArticleHeader Playground smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <h1 class="src-widgets-ArticleHeader-ArticleHeader-module__title--SK4qr">
    Article title
  </h1>
  <img alt="Article image"
       fetchpriority="high"
       width="800"
       height="400"
       decoding="async"
       data-nimg="1"
       class="src-components-Image-Image-module__image--gXjYq"
       srcset="/static/media/_16x9.d50e36d3.png?w=828&amp;q=75 1x, /static/media/_16x9.d50e36d3.png?w=1920&amp;q=75 2x"
       src="/static/media/_16x9.d50e36d3.png?w=1920&amp;q=75"
       style="color: transparent;"
  >
</main>
`;
